rdf_entity.module

```
@@ -187,102 +228,103 @@ function rdf_entity_form_alter(&$form, FormStateInterface $form_state) {
   }
-  $form['#entity_builders'][] = 'rdf_entity_bundle_config';
+  $form['#entity_builders'][] = 'rdf_entity_save_mapping';
 }
 
 /**
- * Store the mapping of fields and rdf properties.
+ * Stores the mapping of base fields and RDF properties.
+ *
+ * @param string $entity_type_id
+ *   The entity type ID.
+ * @param \Drupal\Core\Config\Entity\ConfigEntityInterface $bundle_entity
+ *   The bundle entity being built.
+ * @param array $form
+ *   The form API form render array.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state object.
+ *
+ * @see rdf_entity_form_alter()
  */
-function rdf_entity_bundle_config($entity_type, ConfigEntityInterface $entity, array &$form, FormStateInterface $form_state) {
-  $entity_name = $entity->getEntityType()->getBundleOf();
-  $base_fields = \Drupal::entityManager()->getBaseFieldDefinitions($entity_name);
-  $storage = \Drupal::entityManager()->getStorage($entity_name);
-  $idKey = $storage->getEntityType()->getKey('id');
-  $entity->setThirdPartySetting('rdf_entity', 'entity_id_plugin', $form_state->getValue('entity_id_plugin'));
-  $third_party_settings = [];
-  /** @var \Drupal\Core\Field\BaseFieldDefinition $base_field */
-  foreach ($base_fields as $id => $base_field) {
-    if ($id == $idKey) {
-      continue;
-    }
-    $columns = $base_field->getColumns();
-    foreach ($columns as $column_name => $column) {
-      $third_party_settings[$id][$column_name]['predicate'] = $form_state->getValue('predicate_' . $id . '_' . $column_name);
-      $third_party_settings[$id][$column_name]['format'] = $form_state->getValue('format_' . $id . '_' . $column_name);
-    }
+function rdf_entity_save_mapping(string $entity_type_id, ConfigEntityInterface $bundle_entity, array $form, FormStateInterface $form_state): void {
+  /** @var \Drupal\rdf_entity\RdfEntityMappingInterface $mapping */
+  $mapping = $form_state->get('rdf_entity_mapping');
+  $values = $form_state->getValue('rdf_entity');
+
+  try {
```

Why try? Normally we should have a stable version. Try won't do any good here as it is only 'debugging' it.

If something breaks we should allow it to break because it means we have a bug in our code.

---

OK, let's propagate the exception but I still want to log it.